app/src/main/java/com/sopt/shinmungo/core/designsystem/component/button/BaseRoundedCornerTextButton.kt

```diff
@@ -0,0 +1,98 @@
+package com.sopt.shinmungo.core.designsystem.component.button
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.sopt.shinmungo.core.designsystem.theme.ShinMunGoTheme
+
+/**
+ * 둥근테두리+text 버튼 Base 컴포저블
+ *
+ * @param text 버튼에 들어갈 글자
+ * @param textStyle 글자 스타일
+ * @param textColor 글자 색상
+ * @param isBorderApplied 테두리 적용 여부
+ * @param borderLineColor 테두리 색 (default 값 null)
+ * @param backgroundColor 버튼 배경 색
+ * @param roundedCornerShape 테두리 둥근 정도
+ * @param modifier 수정자
+ */
+
+@Composable
+fun BaseRoundedCornerTextButton(
+    text: String,
+    textStyle: TextStyle,
+    textColor: Color,
+    isBorderApplied: Boolean,
+    borderLineColor: Color? = null,
+    backgroundColor: Color,
+    roundedCornerShape: RoundedCornerShape,
+    modifier: Modifier = Modifier
+) {
+    //테두리 색을 테두리가 있는 경우 테두리 색 적용, 없는 경우 배경색을 적용
+    val lineColor = if (isBorderApplied) borderLineColor else backgroundColor
+
+    Box(
+        modifier = modifier
+            .clip(roundedCornerShape)
+            .background(backgroundColor)
+            .border(width = 1.dp, color = lineColor!!, shape = roundedCornerShape),
```

Comment:
`!!` 연산자는 NPE를 발생시킬 여지가 있기 때문에 왠만하면 사용하지 않는게 좋습니다! 
lineColor와 같이 nullable한 변수는 사용하기 전에 널체크를 하는 것이 좋습니다.

제가 추천드리는 코드는 정답은 아니기 때문에 참고만 해주세요~
```suggestion
        modifier = modifier
            .clip(roundedCornerShape)
            .background(backgroundColor)
            .then(
                  lineColor?.let { Modifier.border(width = 1.dp, color = it, shape = roundedCornerShape) } ?: Modifier
            )
```

Reply:
위에 코드리뷰 달아주신 부분 수정하니 자동으로 수정되었습니다!